src/event_reader_service.rs

```diff
@@ -364,6 +369,130 @@ where
         ))
     }
 
+    pub async fn hard_resync_event(self: Arc<Self>, resync_from_slot: u64) -> Result<()> {
+        let resync_start = match &self
+            .client
+            .get_block(resync_from_slot)
+            .await
+            .map_err(|err| {
+                Error::WrongConfig(format!("Error while getting block by slot: {err:?}"))
+            })?
+            .transactions
+            .first()
+            .ok_or_else(|| Error::WrongConfig("Error while getting tx from block".to_owned()))?
+            .transaction
+        {
+            EncodedTransaction::Json(tx) => Ok(Some(
+                SolanaSignature::from_str(tx.signatures.first().ok_or_else(|| {
+                    Error::WrongConfig("Error while getting tx signature".to_owned())
+                })?)
+                .map_err(|err| {
+                    Error::WrongConfig(format!("Error while signature parsing: {err:?}"))
+                })?,
+            )),
+            other => Err(Error::WrongConfig(format!(
+                "Unexpected format of EncodedTransaction: {other:?}",
+            ))),
+        }?;
```

Comment:
Move out from method to fn

Reply:
Maybe move back to one slot for this method?
 
Because tx in one slot is randomly ordered and I think end-user expected to exactly this slot will be resynced